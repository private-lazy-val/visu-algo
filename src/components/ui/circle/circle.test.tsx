import React from 'react';
import { Circle } from './circle';
import renderer from 'react-test-renderer';
import {ElementStates} from "../../../types/element-states";

describe('Circle Component', () => {
    it('renders without a letter', () => {
        const tree = renderer
            .create(<Circle/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with letters', () => {
        const tree = renderer
            .create(<Circle letter="A"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with head', () => {
        const tree = renderer
            .create(<Circle head="1" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with react-element in head', () => {
        const ReactElement = <span>React Element</span>;
        const tree = renderer
            .create(<Circle head={ReactElement} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with tail', () => {
        const tree = renderer
            .create(<Circle tail="1" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with react-element in tail', () => {
        const ReactElement = <span>React Element</span>;
        const tree = renderer
            .create(<Circle tail={ReactElement} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with index', () => {
        const tree = renderer
            .create(<Circle index={1} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders as small', () => {
        const tree = renderer
            .create(<Circle isSmall={true} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders in default state', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Default} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders in changing state', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Changing} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders in modified state', () => {
        const tree = renderer
            .create(<Circle state={ElementStates.Modified} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});