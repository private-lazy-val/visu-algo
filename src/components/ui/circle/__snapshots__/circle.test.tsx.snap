// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Component renders as small 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;

exports[`Circle Component renders in changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;

exports[`Circle Component renders in default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;

exports[`Circle Component renders in modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;

exports[`Circle Component renders with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            string"
  >
    1
  </div>
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;

exports[`Circle Component renders with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail60 
            element"
  />
</div>
`;

exports[`Circle Component renders with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;

exports[`Circle Component renders with react-element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  >
    <span>
      React Element
    </span>
  </div>
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;

exports[`Circle Component renders with react-element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  >
    <span>
      React Element
    </span>
  </div>
</div>
`;

exports[`Circle Component renders with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            string"
  >
    1
  </div>
</div>
`;

exports[`Circle Component renders without a letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 
            absolute 
            head 
            element"
  />
  <div
    className="circle  default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 
            absolute 
            tail30 
            element"
  />
</div>
`;
